node/src/reactor/main_reactor/control.rs

```diff
@@ -73,6 +73,7 @@ impl MainReactor {
                 }
                 UpgradingInstruction::CatchUp => {
                     info!("Upgrading: switch to CatchUp");
+                    self.last_sync_leap_highest_block_hash = None;
```

Comment:
I don't think we need to ever reset `self.last_sync_leap_highest_block_hash` to `None` (applies to all 4 instances in this file).

Reply:
Updated in ff7100391140362d24541001ab78a2fb2ff88751